lib/api/src/rest/schema.rs

```diff
@@ -132,3 +138,173 @@ pub enum NamedVectorStruct {
     Sparse(segment::data_types::vectors::NamedSparseVector),
     // No support for multi-dense vectors in search
 }
+
+#[derive(Deserialize, Serialize, JsonSchema, Clone, Debug, PartialEq)]
+#[serde(untagged)]
+pub enum OrderByInterface {
+    Key(JsonPath),
+    Struct(OrderBy),
+}
+
+#[derive(Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(untagged, rename_all = "snake_case")]
+pub enum Fusion {
+    /// Reciprocal rank fusion
+    Rrf,
+}
+
+#[derive(Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(untagged)]
+pub enum VectorInput {
+    Id(segment::types::PointIdType),
+    DenseVector(DenseVector),
+    SparseVector(SparseVector),
+    MultiDenseVector(MultiDenseVector),
+}
+
+#[derive(Debug, Serialize, Deserialize, JsonSchema)]
+pub struct QueryRequest {
+    /// Sub-requests to perform first. If present, the query will be performed on the results of the prefetches.
+    #[serde(with = "MaybeOneOrMany")]
+    #[schemars(with = "MaybeOneOrMany<Prefetch>")]
+    pub prefetch: Option<Vec<Prefetch>>,
+
+    /// Query to perform. If missing, returns points ordered by their IDs.
+    pub query: Option<QueryInterface>,
+
+    /// Define which vector to use for querying. If missing, the default vector is used.
+    pub using: Option<String>,
+
+    /// Filter conditions - return only those points that satisfy the specified conditions.
+    pub filter: Option<Filter>,
+
+    /// Search params for when there is no prefetch
+    pub params: Option<SearchParams>,
```

Comment:
what are we doing when the top level `params` is defined AND prefetch is defined?

Reply:
We ignore it